cheetah/track_methods.py

```diff
@@ -96,12 +73,239 @@ def base_rmatrix(
 
     # Rotate the R matrix for skew / vertical magnets
     if torch.any(tilt != 0):
-        R = torch.einsum(
-            "...ij,...jk,...kl->...il", rotation_matrix(-tilt), R, rotation_matrix(tilt)
-        )
+        R = rotation_matrix(-tilt) @ R @ rotation_matrix(tilt)
+
     return R
 
 
+def base_ttensor(
+    length: torch.Tensor,
+    k1: torch.Tensor,
+    k2: torch.Tensor,
+    hx: torch.Tensor,
+    species: Species,
+    tilt: torch.Tensor | None = None,
+    energy: torch.Tensor | None = None,
+) -> torch.Tensor:
+    """
+    Create a second order universal transfer matrix for a beamline element.
+
+    :param length: Length of the element in m.
+    :param k1: Quadrupole strength in 1/m**2.
+    :param k2: Sextupole strength in 1/m**3.
+    :param hx: Curvature (1/radius) of the element in 1/m.
+    :param species: Particle species of the beam.
+    :param tilt: Roation of the element relative to the longitudinal axis in rad.
+    :param energy: Beam energy in eV.
+    :return: Second order transfer matrix for the element.
```

Comment:
Just like in the function name:

```suggestion
    Create a second order universal transfer tensor for a beamline element.

    :param length: Length of the element in m.
    :param k1: Quadrupole strength in 1/m**2.
    :param k2: Sextupole strength in 1/m**3.
    :param hx: Curvature (1/radius) of the element in 1/m.
    :param species: Particle species of the beam.
    :param tilt: Roation of the element relative to the longitudinal axis in rad.
    :param energy: Beam energy in eV.
    :return: Second order transfer tensor for the element.
```